server/routes/loginAuth.js

```diff
@@ -1,47 +1,91 @@
 const router = require("express").Router();
-
-//json web token
+const passport = require("passport");
 const jwt = require("jsonwebtoken");
+const dayjs = require("dayjs");
 
-//google authentication client for token verification
-const { OAuth2Client } = require("google-auth-library");
-const client = new OAuth2Client(process.env.CLIENT_ID);
+require("./loginStrategies/googleStrategy");
 
-router.post("/google", async (req, res) => {
-    const { token } = req.body;
+router.get("/google", passport.authenticate("google", { scope: ["email", "profile"] }));
 
-    try {
-        const ticket = await client.verifyIdToken({
-            idToken: token,
-            audience: process.env.CLIENT_ID,
-        });
+// after successful login with google strategy, user db record will be sent here in the req.user property.
+router.get("/google/callback", passport.authenticate("google", { session: false }), (req, res) => {
+    // Creates a cookie with the user's login information
+    res.cookie("loginCookie", JSON.stringify(req.user), {
+        secure: false,
+        httpOnly: true,
+        expires: dayjs().add(1, "year").toDate(),
+    });
 
-        const { given_name, family_name, email, picture } = ticket.getPayload();
+    // Creates an access jwt token and stores it inside a cookie.
+    const token = jwt.sign({ id: req.user.id }, process.env.TOKEN_SECRET);
+    res.cookie("tokenCookie", token, {
+        secure: false,
+        httpOnly: true,
+    });
+
+    // Both cookies sent to users browser on redirect.
+    res.status(301).redirect("http://localhost:3000");
+});
 
-        //send info to data base using an upsert requests
-        //return created or updated record so user id can be used for JWT.
+// Logs in users using the login information cookie if available.
+router.get("/cookie", (req, res) => {
+    // If the cookie with users login information exists, use it to log in user.
+    if (req.cookies.loginCookie) {
+        // Extracts user id from the cookie with users log in information.
+        const { id } = JSON.parse(req.cookies.loginCookie);
 
-        //testing data of return record
-        const user = {
-            id: "aslkdjhjd7363",
-            first_name: given_name,
-            last_name: family_name,
-            email,
-            picture,
-        };
+        //validate id in the database
 
-        const authToken = jwt.sign({ id: user.id }, process.env.TOKEN_SECRET);
+        // Create an access jwt token using the user id.
+        const token = jwt.sign({ id: id }, process.env.TOKEN_SECRET);
 
-        res.status(200).header("auth-token", authToken).json({
-            status: "success",
-            authToken,
+        // Store created jwt in a cookie.
+        res.cookie("tokenCookie", token, {
```

Comment:
Lets avoid the jwt cookie. We can just store it in a variable on the client side.